content/Snakemake_intro.md

```diff
@@ -12,27 +26,471 @@ Example usecase from GUI -> multiple scripts run manually -> bash script to run
+```
+
+To verify the installation, run:
+```bash
+snakemake -v
+```
+
+You should be able to see the version of Snakemake installed on your system.
 
 ### Snakefile
 
+We can define Snakemake workflows using files called `Snakefile`s. Workflow are defined in terms of rules. Each rule describes input files, outputs and a command or script to be run. Let's have a look at a simple, one step workflow, that will copy the content of one file into another.
+
+```bash
+rule copy:
+    input:
+	    "hello.txt"
+    output: 
+	    "hello_copy.txt"
+    shell:
+	    "cp {input} {output}"
+```
+
+This `Snakefile` defines one input file `hello.txt`, one output file `hello_copy.txt` and a shell command `cp {input} {output}`. The shell command is formatted automatically by Snakemake. The `input` and `output` are substituted using the values provided in the previous section.
+
+Before running, copy the content of the `Snakefile` into the file called `Snakefile` (without an extension) and create `hello.txt` file with some content:
+```bash
+echo "Hello Snakemake!" > hello.txt
+```
+
+To run the `Snakefile`, we run the following command:
+```bash
+snakemake --cores 1
+```
+
+Snakemake executes the `Snakefile` with the one rule in it. The content is copied! Note two things: the `--cores` flag and that we did not specified the `Snakefile`. By default, Snakemake will search for a `Snakefile` in the current directory. We can specify which `Snakefile` to run using the `-s` flag. For example:
+```bash
+snakemake --cores 1 -s Snakefile_2
+```
+
+will run the `Snakefile_2`. Thanks to that, now we can also how multiple `Snakefile`s in one directory. The `--cores` flag tells Snakemake how many resources (cores) from our computer Snakemake can use to execute the workflow. It is mandatory flag, so one cannot omit it.
+
+
+- Definition of the workflow via rules
+- File based approach
+- `expand`
+
 ### Rules
 
-### Dependencies of steps
+A rule do not necessarily have to have only one input or output. If so, we can use indices to refer to each input/output value:
+
+```bash
+rule sorter:
+    input:
+        "hello.txt", 
+        "hello2.txt"
+    output:
+        "sorted_greetings.txt",
+    shell:
+        "sort {input[0]} {input[1]} > {output}"
+```
+
+We can also use named inputs and outputs, and refer to them using their names:
+```bash
+rule sorter:
+    input:
+        a="hello.txt", 
+        b="hello2.txt"
+    output:
+        "sorted_greetings.txt",
+    shell:
+        "sort {input.a} {input.b} > {output}"
+```
 
 ### Wildcards
+In the examples above we used hard-coded filenames to tell the input or output values. To make the workflows more generalized, we can use wildcards. Wildcards are variables that are replacing the actual filenames or any other value, like for example a path. Snakemake resolves them automatically based on either the target file or other input/outputs in a Snakefile. Let's add wildcards to the copy example, so it will work for any text file:
```

Comment:
```suggestion
In the examples above we used hard-coded filenames to define the input and output values. This is not feasible for most workflows dealing with more than a few files. To generalize the workflows, we can use wildcards. Wildcards are variables that replace the actual filenames or any other value, like for example a path. Snakemake resolves them automatically based on either the target file or other input/outputs in the Snakefile. Let's add wildcards to the copy example, so it will work for any text file:
```